tests/end-to-end/ui_smarti/03-resync-smarti.js

```diff
@@ -0,0 +1,634 @@
+/* eslint-env mocha */
```

Comment:
woah! This looks like a really comprehensive test. I didn't verify it line-by line, but it looks really good to me!